auth/src/main/java/com/firebase/ui/auth/KickoffActivity.java

```diff
@@ -1,41 +1,96 @@
 package com.firebase.ui.auth;
 
+import android.app.Dialog;
 import android.content.Context;
+import android.content.DialogInterface;
 import android.content.Intent;
+import android.net.ConnectivityManager;
 import android.os.Bundle;
+import android.util.Log;
 
 import com.firebase.ui.auth.ui.ActivityHelper;
 import com.firebase.ui.auth.ui.AppCompatBase;
 import com.firebase.ui.auth.ui.ExtraConstants;
 import com.firebase.ui.auth.ui.FlowParameters;
 import com.firebase.ui.auth.util.signincontainer.SignInDelegate;
+import com.google.android.gms.common.GoogleApiAvailability;
 
 public class KickoffActivity extends AppCompatBase {
+    private static final String TAG = "KickoffActivity";
+    private static final int RC_PLAY_SERVICES = 1;
+
+    private boolean mIsWaitingForPlayServices = false;
+
+    public static Intent createIntent(Context context, FlowParameters flowParams) {
+        return ActivityHelper.createBaseIntent(context, KickoffActivity.class, flowParams);
+    }
+
     @Override
     protected void onCreate(Bundle savedInstance) {
         super.onCreate(savedInstance);
-        if (savedInstance == null) {
-            SignInDelegate.delegate(this, mActivityHelper.getFlowParams());
+        if (savedInstance == null || !savedInstance.getBoolean(ExtraConstants.HAS_EXISTING_INSTANCE)) {
+            if (!hasNetworkConnection()) {
+                Log.d(TAG, "No network connection");
+                finish(ErrorCodes.NO_NETWORK,
+                       IdpResponse.getErrorCodeIntent(ErrorCodes.NO_NETWORK));
+                return;
+            }
+
+            GoogleApiAvailability apiAvailability = GoogleApiAvailability.getInstance();
```

Comment:
Although you're getting rid of `PlayServicesHelper` I'd still like to centralize all Play services availability logic somewhere.  Even if the class is just a host for a singleton of `GoogleApiAvailability` that's a win for future testing.  Play services has given me a lot of trouble in testing (although you seem to have simplified a lot of it here).